src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -114,17 +114,27 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
         var acceptsMetadata = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>();
         if (acceptsMetadata is not null)
         {
-            var acceptsRequestType = acceptsMetadata.RequestType;
-            var isOptional = acceptsMetadata.IsOptional;
-            var parameterDescription = new ApiParameterDescription
+            // Do not add an implicit body parameter description if a parameter already
+            // exists for something derived from the body's content, such as a form file.
+            var alreadyHasAnyBodyParameter = apiDescription.ParameterDescriptions.Any(x =>
+                x.Source == BindingSource.Body ||
+                x.Source == BindingSource.Form ||
+                x.Source == BindingSource.FormFile);
```

Comment:
> Alternatively I could remove this "default" behaviour for the body by removing the skip for BindingSource.Body and having it dealt with explicitly in CreateApiParameterDescription() instead.

Let's do this for consistency. The main thing that we want to have keep working is if there is no `BindingSource.Body` or `BindingSource.FormFile` parameter by "default" and someone added IAcceptsMetadata explicitly with `.Accepts<TRequest>(...)` or `[Consumes(...)]` indicating they're manually reading the body, we should continue to create this "fake" ApiParameterDescription from IAcceptsMetadata .

If we know we're consuming the body on behalf of the app, we can just assume that the IAcceptsMetadata is what we added by default. If someone tries to add custom IAcceptsMetadata despite that, it's certainly wrong, so no big loss.

Also nit:

```suggestion
            var alreadyHasAnyBodyParameter = apiDescription.ParameterDescriptions.Any(x =>
                x.Source == BindingSource.Body ||
                x.Source == BindingSource.FormFile);
```

I don't see the point in looking for `BindingSources` we'll never create at this point.

Reply:
Fixed nit.

Still thinking about the best way to tackle this, as I've tried a few different ways to remove the default/skip body parameter but they all seemed to break _something_ in the tests.

Reply:
What tests break? I'm okay with changing some of the tests if they're asserting something nonsensical like custom IAcceptsMetadata being used to generate an ApiParameterDescription when we're consuming an application/json request body on behalf of the app. After this change, custom IAcceptsMetadata won't be used to generate an ApiParameterDescription when we're consuming a multipart/form-data on behalf of the app, so at least this is consistent.

Reply:
There were some tests checking there was no body parameter at all, where there was now a default. There were others that would fail because there was no implicit accept metadata where it was expected because the unit tests don't run through the real RequestDelegateFactory code because it's all mocked routing.

For the latter I was torn between changing it to do so to make it more realistic vs. the fact the two implementation details were becoming aware of each other.

I'll come back to it with fresh eyes tomorrow and maybe it'll all click how best to refactor it.